src/mimir_coordinator.py

```diff
@@ -69,6 +69,10 @@ def is_coherent(self) -> bool:
         roles: Iterable[MimirRole] = self._cluster_provider.gather_roles().keys()
         return set(roles).issuperset(MINIMAL_DEPLOYMENT)
 
+    def is_scaled(self) -> bool:
+        """Return True if more than 1 worker are forming the mimir cluster."""
+        return len(list(self._cluster_provider.gather_addresses())) > 1
```

Comment:
Why do we use "gather addresses" (indirect method) instead of looking at the number of units in the relation (direct method) to determine if scaled?

Reply:
Problem here is that only the coordinator (A workload-less charm) will connect to s3, generate the config and then pass it to the workers over the coordinator-worker relation, only the workers will be replicated and in some cases there might be different apps with several units all forming the **mimir-worker-cluster**, so this was the easiest way tbh to find if the workers are scaled. I'm open to better suggestions tho :D

Reply:
Well, that's not strictly true. The coordinator will run nginx.

Reply:
I think the naming of the method is what's causing the confusion here, so I need a small clarification :)

If the correct semantics are expressed in the docstring (which I believe is the case), and we want to know *whether multiple worker units (on any worker app) are related to the coordinator*, then this makes sense (though it could be changed with `units in relation > 1 OR relations > 1`.

In this context, we'd want the method to return `True` even when two single-unit workers are related to the coordinator.

**scaled** here means (IIUC) having multiple workers; every role could still have one unit (so **not scaled**), but the cluster would still be (according to this method).

Am I getting that right?

Reply:
Got the function name changed now to `has_multiple_workers` and I believe I haven't expressed it correctly but mainly even if two single-unit workers are related to the coordinator we would still need mimir to use S3

Reply:
still feels ugly to count the remote units by counting the addresses

more work, but also more transparent, would be to do something like

`sum(len(r.units) for r in self.model.relations["mimir_cluster"])`

with the caveat that you probably need to filter out "this app's units"